mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ExtensionTypesDeclarationEnricher.java

```diff
@@ -137,6 +139,10 @@ private void registerType(ExtensionDeclarer declarer, MetadataType type) {
       public void visitObject(ObjectType objectType) {
         declarer.withType(objectType);
```

Comment:
isn't there another bug in here? If the object type is actually representing a message, should we add the message as an extension type? This doesn't sound right to me at all, but need to analyze this carefully and assess impact in backwards compatibility, check if any other component (studio, DataSense, tooling, etc) is actually depending on the message being described here